kafka_topic_dumper/kafka_client.py

```diff
@@ -185,73 +208,146 @@ def get_messages(self, num_messages_to_consume, max_package_size_in_msgs,
+            def striper(string):
+                match = reg.match(string['Prefix'])
+                if match is not None:
+                    return match.group(1)
+
             prefixes = []
             for response in response_iterator:
-                response_prefixes = [p['Prefix'].replace('-', '')
-                                     for p in response['CommonPrefixes']]
+                response_prefixes = filter(
+                    lambda x: x is not None,
+                    map(striper, response['CommonPrefixes']))
                 prefixes += response_prefixes
+
             dump_prefix = max(prefixes)
             logger.info('Prefix chosen was <{}>'.format(dump_prefix))
 
+        dump_path = prefix + dump_prefix
+
         response_iterator = paginator.paginate(Bucket=bucket_name,
-                                               Prefix=dump_prefix)
+                                               Prefix=dump_path)
         file_names = []
         for response in response_iterator:
             if response['KeyCount'] > 0:
                 file_names = [(f['Key'], f['Size'])
                               for f in response['Contents']]
         file_names.sort()
+        return dump_prefix, file_names
+
+    def _set_test_topic(self, test_id):
+        _, _, end_offsets = self._get_offsets()
+
+        if not end_offsets:
+            msg = 'Can not find offsets for topic <{}>'
+            raise Exception(msg.format(self.topic))
+
+        test_offsets = {}
+
+        for partition, offset in end_offsets.items():
+            test_offsets[partition.partition] = offset
+
+        test_state = {
+            'test_id': test_id,
+            'topic_name': self.topic,
+            'offsets': test_offsets}
+
+        future = self.producer.send(
+            topic=self.dump_state_topic,
+            key=test_id,
+            value=json.dumps(test_state))
+        future.get(timeout=self.timeout)
+        logger.info('State saved')
+
+    def _get_test_state(self, test_id):
+        beginning_offsets, _, end_offsets = (
+            self._get_offsets(topic=self.dump_state_topic))
+        if beginning_offsets:
+            offsets, num_messages_available = self._calculate_offsets(
+                beginning_offsets=beginning_offsets,
+                end_offsets=end_offsets,
+                num_messages_to_consume=1)
+            self._set_offsets(offsets)
+            self.consumer.subscribe(self.dump_state_topic)
+            for message in self._get_messages(num_messages_available):
+                state = json.loads(message[1].decode())
+                if  state.get('topic_name', '') == self.topic and \
+                    state.get('test_id', '') == test_id:
+                    return state.get('offsets', None)
+            return None
+
+    def reload_kafka_server(self, bucket_name, dir_path, dump_prefix=None):
+
+        s3_client = boto3.client('s3')
+
+        test_id, file_names = self._get_file_names(
```

Comment:
Do we really need to query s3 in the case where we'll only reset the offsets?

Reply:
We use it only if we have not the dump prefix already.